hazelcast/src/main/java/com/hazelcast/internal/blackbox/impl/BlackboxImpl.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.blackbox.impl;
+
+import com.hazelcast.internal.blackbox.Blackbox;
+import com.hazelcast.internal.blackbox.DoubleSensorInput;
+import com.hazelcast.internal.blackbox.LongSensorInput;
+import com.hazelcast.internal.blackbox.Sensor;
+import com.hazelcast.internal.blackbox.sensorpacks.ClassLoadingSensorPack;
+import com.hazelcast.internal.blackbox.sensorpacks.GarbageCollectionSensorPack;
+import com.hazelcast.internal.blackbox.sensorpacks.OperatingSystemSensorPack;
+import com.hazelcast.internal.blackbox.sensorpacks.RuntimeSensorPack;
+import com.hazelcast.internal.blackbox.sensorpacks.ThreadSensorPack;
+import com.hazelcast.logging.ILogger;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.hazelcast.util.Preconditions.checkNotNull;
+import static java.lang.String.format;
+
+/**
+ * The {@link Blackbox} implementation.
+ *
+ * Interesting read:
+ * https://source.android.com/devices/sensors/sensor-types.html
+ */
+public class BlackboxImpl implements Blackbox {
```

Comment:
ok. I'm fine with that. 
I thought you weren't a fan of interfaces when only a single implemented is expected.